db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/post_provision.rb

```diff
@@ -1 +1,115 @@
+require 'net/ssh'
 
+def rhsm_user
+  $evm.root['automation_task'].automation_request.options[:attrs][:rhsm_user]
+end
+
+def rhsm_password
+  $evm.root['automation_task'].automation_request.options[:attrs][:rhsm_password]
+end
+
+def rhsm_sku
+  $evm.root['automation_task'].automation_request.options[:attrs][:rhsm_sku]
+end
+
+# need to check if clients need to add pool or just employees
+# need to have activation key command as well
+COMMANDS = [
+  "subscription-manager register --username=#{rhsm_user}  --password=#{rhsm_password}",
+  "subscription-manager attach --pool=#{rhsm_pool_id}",
+  "subscription-manager repos --disable=\"*\"",
+  "subscription-manager repos --enable=\"rhel-7-server-rh-common-rpms\" --enable=\"rhel-7-server-rpms\" --enable=\"rhel-7-server-extras-rpms\" --enable=\"rhel-7-server-ose-3.1-rpms\""
+].freeze
+
+def find_vm_by_tag(tag)
+  $evm.vmdb(:vm).find_tagged_with(:any => tag, :ns => "*")
+end
+
+def tagged_vms
+  nodes_tag = "/managed/deploy/" + "node" + "_#{$evm.root['automation_task'][:id]}"
+  masters_tag = "/managed/deploy/" + "master" + "_#{$evm.root['automation_task'][:id]}"
+  return find_vm_by_tag(masters_tag), find_vm_by_tag(nodes_tag)
+end
+
+def assign_vms_to_deployment_nodes(masters, nodes)
+  deployment = $evm.vmdb(:container_deployment).find($evm.root['automation_task'].automation_request.options[:attrs][:deployment_id])
+  deployment_master = masters.first
+  deployment.assign_contianer_deployment_node(deployment_master.id, :deployment_master)
+  masters.delete(deployment_master)
+  masters.each do |master|
+    deployment.assign_contianer_deployment_node(master.id, :masters)
+  end
+  nodes.each do |node|
+    deployment.assign_contianer_deployment_node(node.id, :nodes)
+  end
+  $evm.root['automation_task'].automation_request.options[:attrs][:inventory] = deployment.regenerate_ansible_inventory
+  $evm.root['automation_task'].automation_request.options[:attrs][:rhel_subscribe_inventory] =  deployment.regenerate_ansible__subscription_inventory
+end
+
+def missing_subscription_fields?
+  rhsm_user.nil? || rhsm_password.nil? || rhsm_sku.nil?
+end
+
+def subscribe_if_rhel
+  Net::SSH.start($evm.root['deployment_master'], $evm.root['user'], :paranoid => false, :forward_agent => true,
+                 :key_data => $evm.root['private_key']) do |ssh|
+    $evm.log(:info, "Connected to deployment master , ip address: #{$evm.root['deployment_master']} Checking if RHEL")
+    type = ssh.exec!("cat /etc/*release")
+    subscrption_needed = type.include? "REDHAT_SUPPORT_PRODUCT=\"Red Hat Enterprise Linux\""
+    if subscrption_needed && missing_subscription_fields?
+      $evm.root['ae_result'] = "error"
+      $evm.log(:info, "Missing credentials for rhn subscription")
+      $evm.root['automation_task'].message = "Missing credentials for rhn subscription"
+    elsif subscrption_needed
+      $evm.log(:info, "RHN subscribing starting")
+      COMMANDS.each do |cmd|
+        ssh.exec!(cmd)
+      end
```

Comment:
where do you verify the commands execution was successful? 
you can look at ssh_exec! implementation in pre_deployment state.

also, IINM you might present a wrong message in case one of the commands fail and not the ssh to deployment master.